packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1993,6 +1993,7 @@ function updateDeferredValueImpl<T>(hook: Hook, prevValue: T, value: T): T {
       markWorkInProgressReceivedUpdate();
     }
 
+    hook.memoizedState = value;
```

Comment:
Is it possible for an urgent update to interrupt this non urgent update before it commits? Would this leak the new value into the urgent update that follows?

Reply:
If there's an interrupting we throw out all the work-in-progress fibers and hooks. `hook` on this line refers to a work-in-progress.

Reply:
Alas, it's JavaScript so there's no borrow checker. You are the borrow checker.